dnnsoftware/Dnn.Platform
# Add UI to Manage Google Analytic Tracking Code

Closes #2129 

## Summary
Added a new Connector extension project to the solution called DNN.GoogleAnalyticsConnector.

The connector adds the previously removed fields for managing the Google Analytic fields that were previously exposed in the Platform in prior versions of DNN. The fields should match the fields that were in DNN 8.0.4.

The connector also adds a new button for deactivating the analytic codes. This button is visible only when an analytics code is saved. Behind the scenes, this behaviour only clears the fields for the GA provider which is turn prevents the provider from writing code to the page. This isn't, however, a true deactivation of the GA Http Module and site analytics config.

Validation was added to verify that the Google Analytics tracking code is not empty.

## How to Test

On an existing site that is using the core Google Analytics provider, you should be able to view and edit the previously defined tracking code settings. These settings are stored at the root of each portal folder within the GoogleAnalytics.config file.

On a new installation, you should be able to fill out the tracking code details and have them reflected within the page source after saving the screen. Additionally, the GoogleAnalytics.config file should now be visible on the root of the portal in question with the proper settings added to it.

After saving the tracking code, try deleting it and re-adding,etc.

## Known Issues

1. What I perceive to be a workaround for the showing/hiding the Delete Connection button was implemented due to some shortcomings in the connectors extension.  This has been indicated with comments in both the C# and JS.

2. Pressing cancel after the connector has been saved will make it appear that it is no longer saved.  This appears to be an issue with the core connectors extension. Another issue will be created to review the connectors extension in general.
